### PR TITLE
✨Added nameType prop

### PR DESCRIPTION
**Pull request by anto90fg**
#### Description

I thought of adding the possibility for the user to be able to choose whether to display in the ProductGiftName the skuName or the productName, since as in my case the skuName is valued by taking the product name and associating it with the sku identifier. 
By default I have left the standard behaviour of the component unchanged by setting the relative prop to false.

#### Screenshots 
This is the default behaviour, in blue the product name and in red the sku id.
image
Instead this is when the new prop is set to true, so only the product name is displayed.
image

#### How to test it?

[Product link](https://giftproductname--itwhirlpool.myvtex.com/lavatrice-a-libera-installazione-a-carica-dallalto-whirlpool-70-kg-tdlr-7221bs-itn-859991613560/p)

##### Version
Version: 0.2.1
Proposed version: 0.3.0

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -139,6 +139,7 @@ You can configure the string received by the `text` prop using the following var
 | Prop name           | Type      | Description                                                               | Default value |
 | ------------------- | --------- | ------------------------------------------------------------------------- | ------------- |
 | `linkToProductPage` | `Boolean` | Whether or not the `gift-name` block should be a link to the gift's product page. | `false`       |
+| `showProductName` | `Boolean` | Whether the `gift-name` block should show the product name instead the sku name | `false`       |
```
Comment (icazevedo) [inline docs/README.md]: We usually avoid boolean props to better allow the evolution of the framework and avoid creating multiple props that are related to the same thing.

What I suggest for you to do is to change this prop to accept two possible values: `'productName'` or `'skuName'`, defaulting to `'skuName'` of course.

**Review by icazevedo — CHANGES_REQUESTED** [review]
Thanks for the PR! Please take a look at the suggestion I've left. Also, your lint definitions seem to be different from ours. Please fix that too! We don't use `;` or `"` for example.

**Comment by anto90fg** [comment]
> Thanks for the PR! Please take a look at the suggestion I've left. Also, your lint definitions seem to be different from ours. Please fix that too! We don't use `;` or `"` for example.

Fixed ;)

File: docs/README.md (hunk 1 above)
Comment (anto90fg) [reply docs/README.md]: As requested, I have changed the type of prop, transforming it into an enumerate with the two suggested parameters and using skuName as the default one :)

File: react/ProductGiftName.tsx [hunk 2]
```diff
@@ -62,6 +76,12 @@ ProductGiftName.schema = {
       description: messages.linkToProductDescription.id,
       type: 'boolean',
     },
+    nameType: {
+      default: false,
+      title: messages.nameType.id,
+      description: messages.nameType.id,
+      type: 'boolean',
+    },
```
Comment (victorhmp) [inline react/ProductGiftName.tsx]: This definition would result in a boolean field in the Site Editor. I think you meant to declare an enum here, right?

**Review by victorhmp — CHANGES_REQUESTED** [review]

File: react/ProductGiftName.tsx (hunk 2 above)
Comment (victorhmp) [reply react/ProductGiftName.tsx]: Here's an example: https://github.com/vtex-apps/menu/blob/master/store/contentSchemas.json#L63

Comment (anto90fg) [reply react/ProductGiftName.tsx]: Yes it was my mistake, I fixed it.
@victorhmp

File: react/ProductGiftName.tsx [hunk 3]
```diff
@@ -62,6 +76,20 @@ ProductGiftName.schema = {
...
+      "type": "string",
+      "enum": [
+        "skuName",
+        "productName"
+      ],
+      "enumNames": [
+        "skuName",
+        "productName"
+      ],
+      "default": "skuName"
```
Comment (victorhmp) [inline react/ProductGiftName.tsx]:
```suggestion
      type: "string",
      enum: [
        "skuName",
        "productName"
      ],
      enumNames: [
        "skuName",
        "productName"
      ],
      default: "skuName"
```

**Review by victorhmp — CHANGES_REQUESTED** [review]

File: react/ProductGiftName.tsx (hunk 3 above)
Comment (anto90fg) [reply react/ProductGiftName.tsx]: Done! :)

**Review by icazevedo — APPROVED** [review]

File: docs/README.md [hunk 4]
```diff
@@ -139,6 +139,7 @@ You can configure the string received by the `text` prop using the following var
 | Prop name           | Type      | Description                                                               | Default value |
 | ------------------- | --------- | ------------------------------------------------------------------------- | ------------- |
 | `linkToProductPage` | `Boolean` | Whether or not the `gift-name` block should be a link to the gift's product page. | `false`       |
+| `nameType` | `enum` | By default the sku name associated with the product is displayed, otherwise you can choose to display the product name using `productName` as value. | `skuName`       |
```
Comment (guerreirobeatriz) [inline docs/README.md]:
```suggestion
| `nameType` | `enum` | Name type to be displayed alongside the gift. Possible values are:  `productName` (displays the gift's product name) and `skuName` (displays the gift's SKU name). | `skuName` |
```

**Review by guerreirobeatriz — CHANGES_REQUESTED** [review]

**Review by igorbrasileiro — APPROVED** [review]
LGTM

**Review by victorhmp — APPROVED** [review]

**Pull request merged**